src/ModelContextProtocol/Utils/Json/JsonStringEnumConverterWithEnumMemberAttrSupport.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+// NOTE:
+// This is a temporary workaround for lack of System.Text.Json's JsonStringEnumConverter<T>
+// 9.x support for JsonStringEnumMemberNameAttribute. Once all builds use the System.Text.Json 9.x
+// version, this whoe file can be removed.
+
+namespace System.Text.Json.Serialization;
+
+internal sealed class JsonStringEnumConverterWithEnumMemberAttrSupport<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)] TEnum> :
+    JsonStringEnumConverter<TEnum> where TEnum : struct, Enum
+{
+    public JsonStringEnumConverterWithEnumMemberAttrSupport() :
+        base(namingPolicy: ResolveNamingPolicy())
```

Comment:
I think we can get rid of the naming policy if compiling `NET9_0_OR_GREATER` no?